hooks/useNotificationManager.ts

```diff
@@ -0,0 +1,105 @@
+import { useAccount, useProvider } from '@starknet-react/core';
+import { useAtom } from 'jotai';
+import { atomWithStorage } from 'jotai/utils';
+import { useEffect } from 'react';
+import { hexToDecimal } from '../utils/feltService';
+import { NotificationType } from '../utils/constants';
+import {
+  RejectedTransactionReceiptResponse,
+  RevertedTransactionReceiptResponse,
+  TransactionType,
+} from 'starknet';
+
+export type NotificationData = TransactionData;
+
+export type SIDNotification<T> = {
+  address?: string; // decimal address
+  timestamp: number;
+  subtext: string;
+  type: NotificationType;
+  data: T;
+};
+
+export type TransactionData = {
+  type: TransactionType;
+  hash: string;
+  status: 'pending' | 'success' | 'error';
+  txStatus?:
+    | 'NOT_RECEIVED'
+    | 'RECEIVED'
+    | 'ACCEPTED_ON_L2'
+    | 'ACCEPTED_ON_L1'
+    | 'REJECTED'
+    | 'REVERTED';
+};
+
+const notificationsAtom = atomWithStorage<SIDNotification<NotificationData>[]>(
+  'userNotifications_SID',
+  []
+);
+
+export function useNotificationManager() {
+  const { provider } = useProvider();
+  const { address } = useAccount();
+  const [notifications, setNotifications] = useAtom(notificationsAtom);
```

Comment:
Are you sure you need jotai and useAtom ? Can't you do the same with useState from React ?